core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerUtils.scala

```diff
@@ -340,4 +347,24 @@ private[mesos] trait MesosSchedulerUtils extends Logging {
     sc.conf.getTimeAsSeconds("spark.mesos.rejectOfferDurationForUnmetConstraints", "120s")
   }
 
+  protected def getExecutorRelatedPort(sc: SparkContext, configProperty: String): Int = {
+    // either: spark.blockManager.port or spark.executor.port
+    sc.conf.getInt(s"$configProperty", 0)
+  }
+
+  protected def checkPorts(sc: SparkContext, ports: List[(Long, Long)]): Boolean = {
```

Comment:
yeah, the existing code is not perfect, especially Mesos integration code which really lacks documentation. In general I would err on adding more comments than not adding them at all, since the goal of adding comments is to make it easy for someone new to follow. Even just a sentence on what the return value means is better than nothing.